### PR TITLE
Fixed radius units in seeding

### PR DESCRIPTION
**Pull request by bschmookler**
This fixes a unit in the calculation of the track seed momentum.

### CODE REVIEW

File: src/algorithms/tracking/TrackSeeding.cc [hunk 1]
```diff
@@ -94,7 +94,7 @@ std::vector<edm4eic::TrackParameters*> eicrecon::TrackSeeding::makeTrackParams(S
       if(theta < 0)
 	{ theta += M_PI; }
       float eta = -log(tan(theta/2.));
-      float pt = 0.3 * R * (m_cfg.m_bFieldInZ * 1000) / 100.;
+      float pt = 0.3 * (R / 1000.) * (m_cfg.m_bFieldInZ * 1000.);
```
Comment (wdconinc) [inline src/algorithms/tracking/TrackSeeding.cc:97]: e.g.
```suggestion
      float pt = 0.3 * (R / dd4hep::mm) * (m_cfg.m_bFieldInZ * Acts::UnitConstants::mm);
```

Comment (wdconinc) [reply src/algorithms/tracking/TrackSeeding.cc:97]: which is wrong. just an example.

Comment (bschmookler) [reply src/algorithms/tracking/TrackSeeding.cc:97]: Ok, I can do that. But I'm a bit confused about the position being used in the simulation now. If I check the units in dd4hep, I see the following:

```
root -l
root [0] #include <Evaluator/DD4hepUnits.h>
root [1] dd4hep::cm
(const double) 1.0000000
root [2] dd4hep::mm
(const double) 0.10000000
```

This has cm set to 1. However, in the standard DD4HEP and EICRecon output ROOT files, unless I'm misreading the data, it looks like the position units for the tracker hits are given in mm. So, how would I use the above to convert from mm to m?

Comment (bschmookler) [reply src/algorithms/tracking/TrackSeeding.cc:97]: I can of course change the 0.3 factor.

Comment (DraTeots) [reply src/algorithms/tracking/TrackSeeding.cc:97]: Unfortunately almost each software has to define its own units header. For now (thanks Wouter):

EDM4hep: GeV, mm, ns (also juggler units)
DD4hep: GeV, cm, s
ROOT: GeV, cm, s
Acts: GeV, mm, mm
Geant4: MeV, mm, ns

So as you see, EDM4hep i.e. our data model uses mm and DD4Hep as well as ROOT geometry uses cm.

File: src/algorithms/tracking/TrackSeeding.cc [hunk 2]
```diff
@@ -94,7 +94,7 @@ std::vector<edm4eic::TrackParameters*> eicrecon::TrackSeeding::makeTrackParams(S
       if(theta < 0)
 	{ theta += M_PI; }
       float eta = -log(tan(theta/2.));
-      float pt = 0.3 * R * (m_cfg.m_bFieldInZ * 1000) / 100.;
+      float pt = 0.3 * R * m_cfg.m_bFieldInZ; // pt [GeV] = 0.3 * R[mm] * B[kT]
```
[the remark this answers is not in the document]
Comment (wdconinc) [reply src/algorithms/tracking/TrackSeeding.cc:97]: It's proper definition, that is. Not sure what it is here.

[the remark this answers is not in the document]
Comment (bschmookler) [reply src/algorithms/tracking/TrackSeeding.cc:97]: That would be fine. 

But I'm looking a bit closer at the ACTS magnetic field units: 

https://github.com/acts-project/acts/blob/main/Core/include/Acts/Definitions/Units.hpp#L33
https://github.com/acts-project/acts/blob/main/Core/include/Acts/Definitions/Units.hpp#L191

We currently set the magnetic field to `0.0017 kT` here: https://github.com/eic/EICrecon/blob/main/src/algorithms/tracking/OrthogonalTrackSeedingConfig.h#L35

And then pass that to the ACTS seeder. (And use it in the line I modified in this pull request.) Based on the ACTS default units, I wonder if that parameter should be modified from `kT` to `GeV/(e*mm) `before passing it to the seeder.

[the remark this answers is not in the document]
Comment (bschmookler) [reply src/algorithms/tracking/TrackSeeding.cc:97]: Yes. Let me do that and test that the seeding codes reconstructs a reasonable q/p.

[the remark this answers is not in the document]
Comment (bschmookler) [reply src/algorithms/tracking/TrackSeeding.cc]: I made the changes and I think it is ready to be merged. 

I generated some events and tested as follows:

```
root [1] events->SetAlias("P_gen","sqrt(MCParticles.momentum.x*MCParticles.momentum.x+MCParticles.momentum.y*MCParticles.momentum.y+MCParticles.momentum.z*MCParticles.momentum.z)")
(bool) true
root [2] events->SetAlias("P_seed","CentralTrackSeedingResults.charge/CentralTrackSeedingResults.qOverP")
(bool) true
root [4] events->Draw("(P_seed-P_gen)/P_gen","MCParticles.generatorStatus==1")
```

pic1

Results look reasonable -- few percent momentum resolution for the seeds.

**Review by wdconinc — APPROVED** [review]
Two more comment for your consideration in this or other PR, but approved.

**Review by wdconinc — APPROVED** [review]

File: src/algorithms/tracking/OrthogonalTrackSeedingConfig.h [hunk 3]
```diff
@@ -32,7 +32,7 @@ namespace eicrecon {
     float m_sigmaScattering = 5;
     float m_radLengthPerSeed = 0.1;
     float m_minPt = 100.; // MeV
-    float m_bFieldInZ = 0.0017; //kTesla
+    float m_bFieldInZ = 1.7 * Acts::UnitConstants::T; // GeV/(e*mm)
```
[the remark this answers is not in the document]
Comment (bschmookler) [reply src/algorithms/tracking/OrthogonalTrackSeedingConfig.h]: Done.

File: src/algorithms/tracking/OrthogonalTrackSeedingConfig.h [hunk 4]
```diff
@@ -32,7 +32,7 @@ namespace eicrecon {
     float m_sigmaScattering = 5;
     float m_radLengthPerSeed = 0.1;
     float m_minPt = 100.; // MeV
```
[the remark this answers is not in the document]
Comment (bschmookler) [reply src/algorithms/tracking/OrthogonalTrackSeedingConfig.h]: I did this. We will make a separate commit once the seed parameters are optimized.

**Pull request merged**